src/coreclr/inc/clrconfigvalues.h

```diff
@@ -280,6 +280,13 @@ RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_HeapVerify, W("HeapVerify"), 0, "When set v
 RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_GCNumaAware, W("GCNumaAware"), 1, "Specifies if to enable GC NUMA aware")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_GCCpuGroup, W("GCCpuGroup"), 0, "Specifies if to enable GC to support CPU groups")
 RETAIL_CONFIG_STRING_INFO(EXTERNAL_GCName, W("GCName"), "")
+/**
+ * This flag allows us to force the runtime to use global allocation context on Windows x86/amd64 instead of thread allocation context just for testing purpose.
+ * The flag is unsafe for a subtle reason. Although the access to the g_global_alloc_context is protected under a lock. The implementation of
+ * that lock in the JIT helpers are not multi-core safe (in particular, it used and inc instruction without using the LOCK prefix). This is
+ * only useful for ad-hoc testing.
+ */
+CONFIG_DWORD_INFO(INTERNAL_GCUseGlobalAllocationContext, W("GCUseGlobalAllocationContext"), 0, "Force using the global allocation context for testing only")
```

Comment:
Is using `DOTNET_PROCESSOR_COUNT` not sufficient to give you the behavior?

Reply:
Nope, it doesn't change `g_SystemInfo.dwProcessorCount` and won't impact the behavior at all.

Reply:
Separately, if that defines the number of compute cores available, should we be using the global alloc context in that scneario?

Reply:
As the comment indicates, the implementation of global allocation context is not multi-core safe, and it is by design for performance. Unless that setting will restrict the process from actually using more than one core. This is not safe.

Reply:
That setting is meant to restrict how many cores a dotnet process can use, so it should be safe